sample-apps/react-native/dogfood/src/components/Button.tsx

```diff
@@ -24,6 +24,9 @@ export const Button = ({
   titleStyle,
   ...rest
 }: ButtonPropTypes) => {
+  const { theme } = useTheme();
+  const styles = getStyles(theme);
```

Comment:
@santhoshvai what do you think of this approach passing the theme to `getStyles` so that all `appTheme` usages in `dogfood` app are replaced with proper usage of the theme so that we can more easily switch theme on app level dinamically? If this is ok, I can replace all `appTheme` usages in the `dogfood` app.

Reply:
Chiming-in here: I think it would be better if `getStyles` is turned into `useStyles()` that internally memoizes the returned stylesheet. Something like this:

```ts
const useStyles = () => {
  const theme = useTheme();
  return useMemo(() => {
    return StyleSheet.create({
      button: {
        //backgroundColor: theme.colors.primary,
        backgroundColor: theme.getValue('button', 'backgroundColor'),
        justifyContent: 'center',
        borderRadius: 8,
        height: BUTTON_HEIGHT,
        paddingHorizontal: theme.variants.spacingSizes.md,
      },
      buttonText: {
        color: theme.colors.static_white,
        fontWeight: theme.typefaces.heading6.fontWeight,
        textAlign: 'center',
        fontSize: 17,
      },
      disabledButtonStyle: {
        backgroundColor: theme.colors.disabled,
      },
    );
  }, [theme]);
}
```

This hook should be implemented for every component that uses these theme variables.

Reply:
Side note: StyleSheet.create is for compile time type checking.Also you can look at Stylesheet.compose if you want to merge styles with compile time type checking.